horovod/spark/keras/remote.py

```diff
@@ -112,10 +112,12 @@ def train(serialized_model, train_rows, val_rows, avg_row_size):
 
         pin_gpu(hvd, tf, k)
 
-        if not user_shuffle_buffer_size:
+        # If user specifies any user_shuffle_buffer_size (even 0), we should honor it.
+        if user_shuffle_buffer_size is None:
             shuffle_buffer_size = calculate_shuffle_buffer_size(
                 hvd, avg_row_size, train_rows / hvd.size())
         else:
+            assert user_shuffle_buffer_size >= 0, "user_shuffle_buffer_size cannot be negative!"
```

Comment:
Should prefer to raise a ValueError instead of asserting.

Reply:
Will do